python_modules/dagster/dagster_tests/core_tests/storage_tests/test_fs_io_manager.py

```diff
@@ -361,3 +366,57 @@ def four(inp):
         assert os.path.isfile(filepath_b)
         with open(filepath_b, "rb") as read_obj:
             assert pickle.load(read_obj) == 4
+
+
+def test_fs_io_manager_none():
+    with tempfile.TemporaryDirectory() as tmpdir_path:
+        io_manager_def = fs_io_manager.configured({"base_dir": tmpdir_path})
+
+        @asset
+        def asset1() -> None:
+            pass
+
+        @asset(non_argument_deps={"asset1"})
+        def asset2() -> None:
+            pass
+
+        result = materialize(
+            with_resources([asset1, asset2], resource_defs={"io_manager": io_manager_def})
+        )
+
+        assert not os.path.exists(os.path.join(tmpdir_path, "asset1"))
+        assert not os.path.exists(os.path.join(tmpdir_path, "asset2"))
+        handled_output_events = list(
+            filter(lambda evt: evt.is_handled_output, result.all_node_events)
+        )
+        assert len(handled_output_events) == 2
+
+        for event in handled_output_events:
+            assert len(event.event_specific_data.metadata_entries) == 0
+
+
+def test_fs_io_manager_ops_none():
+    with tempfile.TemporaryDirectory() as tmpdir_path:
+        io_manager_def = fs_io_manager.configured({"base_dir": tmpdir_path})
+
+        @op
+        def op1() -> None:
+            pass
+
+        @op(ins={"abc": In(Nothing)})
+        def op2() -> None:
+            pass
+
+        @job(resource_defs={"io_manager": io_manager_def})
+        def job1():
+            op2(op1())
+
+        result = job1.execute_in_process()
+
+        handled_output_events = list(
+            filter(lambda evt: evt.is_handled_output, result.all_node_events)
+        )
+        assert len(handled_output_events) == 2
+
+        for event in handled_output_events:
+            assert len(event.event_specific_data.metadata_entries) == 0
```

Comment:
I know that the implementation bypasses this, but for posterity's sake it might be good to have a test that optional types still get handled properly vs None/Nothing types.